### ISSUE TITLE
Dismiss displaying outgoins across atomspaces

### ISSUE THREAD

**Issue by AmeBel** [opencog/atomspace#1640]
### Description
When a link is created in a child atomspace between an atom only found in it and another atom only found in its parent, the atom in the parent disappears from the parent atomspace.

### Steps to reproduce
``` scheme
>> guile
GNU Guile 2.2.3
Copyright (C) 1995-2017 Free Software Foundation, Inc.

Guile comes with ABSOLUTELY NO WARRANTY; for details type `,show w'.
This program is free software, and you are welcome to redistribute it
under certain conditions; type `,show c' for details.

Enter `,help' for help.
scheme@(guile-user)> (use-modules (opencog))
scheme@(guile-user)> (define parent-as (cog-atomspace))
scheme@(guile-user)> (define child-as (cog-new-atomspace parent-as))
scheme@(guile-user)> (define p (Node "p"))
scheme@(guile-user)> (cog-prt-atomspace)
(Node "p")
scheme@(guile-user)> (cog-set-atomspace! child-as )
$1 = #<atomspace 0x55555555541f12b5>
scheme@(guile-user)> (cog-prt-atomspace )
(Node "p")
scheme@(guile-user)> (define c (Node "c"))
scheme@(guile-user)> (cog-prt-atomspace )
(Node "p")
(Node "c")
scheme@(guile-user)> (cog-set-atomspace! parent-as )
$2 = #<atomspace 0x5555555554198ac5>
scheme@(guile-user)> (cog-prt-atomspace )
(Node "p")
scheme@(guile-user)> (cog-set-atomspace! child-as )
$3 = #<atomspace 0x55555555541f12b5>
scheme@(guile-user)> (cog-prt-atomspace )
(Node "p")
(Node "c")
scheme@(guile-user)> (Link p c)
$4 = (Link
   (Node "p")
   (Node "c")
)

scheme@(guile-user)> (cog-prt-atomspace )
(Link
   (Node "p")
   (Node "c")
)
scheme@(guile-user)> (cog-set-atomspace! parent-as )
$5 = #<atomspace 0x5555555554198ac5>
scheme@(guile-user)> (cog-prt-atomspace )
scheme@(guile-user)> (cog-set-atomspace! child-as )
$6 = #<atomspace 0x55555555541f12b5>
scheme@(guile-user)> (cog-prt-atomspace )
(Link
   (Node "p")
   (Node "c")
)
```

### Repository details
cogutis sha = `a0d64bdd29d9e4ed5baee664c606fa254174791a Merge pull request #105 from edguy3/while-semi-error`

atomspace sha = `faddec5fafcce3dc12f41c16084284fc634e1c29 Merge pull request #1635 from linas/sql`

### System details
* `4.4.0-119-generic #143~14.04.1-Ubuntu SMP Mon Apr 2 18:04:36 UTC 2018 x86_64 GNU/Linux`
* `g++ (Ubuntu 4.8.5-2ubuntu1~14.04.1) 4.8.5`

**Comment by ngeiswei** [comment]
This bug could totally belong to https://wiki.opencog.org/w/Quick_tasks if nobody wants to fix it right now I'll add it there.

**Comment by linas** [comment]
fixed by #1675
